pkg/otelcgen/exporters.go

```diff
@@ -10,12 +10,13 @@ var (
 
 func (c *Config) buildExporters() map[component.ID]component.Config {
 	serverConfig := &ServerConfig{
-		Endpoint:   c.buildExportersEndpoint(),
-		TLSSetting: &TLSSetting{},
+		Endpoint: c.buildExportersEndpoint(),
 	}
 
 	if c.caFile != "" {
-		serverConfig.TLSSetting.CAFile = c.caFile
+		serverConfig.TLSSetting = &TLSSetting{
```

Comment:
it can panic, if TLSSetting is not set, AFAIK, since it pointer.

Reply:
potentially we can update test/ or write new one.

Reply:
We tried that with nil, it seems to work fine. It is used to avoid `tls: {}` in the receiver as well as in the exporter configuration.

Reply:
Its the same as here: https://github.com/Dynatrace/dynatrace-operator/blob/6ce9df993a1e259a560145d2a7375ec1191fc1a7/pkg/otelcgen/otelcgen.go#L126

Reply:
And the parent struct has an `omitempty` annotation: https://github.com/Dynatrace/dynatrace-operator/blob/6ce9df993a1e259a560145d2a7375ec1191fc1a7/pkg/otelcgen/otelcgen.go#L41

Reply:
or i forgot that i did it.